datafeed/validations.go

```diff
@@ -0,0 +1,98 @@
+package datafeed
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/datafeed/proto"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+)
+
+type verifyApiResponse struct {
+	Outcome   int32
+	Timestamp int64
+}
+
+func (d *DataFeed) verifyMarketOutcome(payload *proto.DataFeedReport, verifyApiHost string) (bool, error) {
+	requestURL := fmt.Sprintf("%s/api/outcome/%s", verifyApiHost, payload.MarketHash)
+	response, err := http.Get(requestURL)
+
+	if err != nil {
+		d.logger.Error("[MARKET-VERIFICATION] Failed to verify market outcome with server error", "error", err)
+		return false, err
+	}
+
+	defer response.Body.Close()
+	body, parseErr := ioutil.ReadAll(response.Body)
+
+	if parseErr != nil {
+		d.logger.Error("[MARKET-VERIFICATION] Failed to parse response", "parseError", parseErr)
+		return false, parseErr
+	}
+
+	if response.StatusCode != 200 {
+		d.logger.Error("[MARKET-VERIFICATION] Failed to verify market with invalid outcome/payload", "market", payload.MarketHash, "outcome", payload.Outcome, "parsedBody", body)
+		return false, fmt.Errorf("Failed to verify market with invalid outcome/payload")
+	}
+
+	var data verifyApiResponse
+	var marshalErr = json.Unmarshal(body, &data)
+
+	if marshalErr != nil {
+		d.logger.Error("[MARKET-VERIFICATION] Failed to unmarshal outcome", "body", body, "parseError", marshalErr)
+		return false, marshalErr
+	}
+
+	if data.Outcome == payload.Outcome {
+		return true, nil
```

Comment:
I don't think you ever actually do anything with this `true` return value. 

You could have this function signature just return `error` and just have the final lines be:

```
if data.Outcome != payload.Outcome {
  return fmt.Errorf("ERROR MESSAGE HERE")
}
```